examples/kuka_iiwa_arm/kuka_simulation.cc

```diff
@@ -0,0 +1,228 @@
+
+  drake::multibody::MultibodyPlant<double>* world_plant =
+      owned_world_plant.get();
+  drake::multibody::MultibodyPlant<double>* controller_plant =
+      owned_controller_plant.get();
+  drake::geometry::SceneGraph<double>* scene_graph =
+      owned_scene_graph.get();
+  world_plant->RegisterAsSourceForSceneGraph(scene_graph);
+
+  // Get the Iiwa model. TODO: grab this from pegged drake libraries
+  const char* kModelPath =
+      "drake/manipulation/models/iiwa_description/iiwa7/iiwa7_no_collision.sdf";
+  const std::string kuka_urdf = drake::FindResourceOrThrow(kModelPath);
+  const auto X_WI = RigidTransform<double>::Identity();
+
+  // Add the Iiwa model to the world model
+  drake::multibody::Parser world_plant_parser(world_plant);
+  const drake::multibody::ModelInstanceIndex iiwa_model =
+      world_plant_parser.AddModelFromFile(kuka_urdf, "iiwa");
+  world_plant->WeldFrames(owned_world_plant->world_frame(),
+      owned_world_plant->GetFrameByName("iiwa_link_0", iiwa_model), X_WI);
+
+  // Create and add a plant to the controller-specific model
+  drake::multibody::Parser controller_plant_parser(controller_plant);
+  const auto controller_iiwa_model = controller_plant_parser.AddModelFromFile(
+      kuka_urdf, "iiwa");
+  owned_controller_plant->WeldFrames(owned_controller_plant->world_frame(),
+      owned_controller_plant->GetFrameByName("iiwa_link_0", controller_iiwa_model),
+      X_WI);
+
+  // Finalize the plants to begin adding them to a system
+  owned_controller_plant->Finalize();
+  world_plant->Finalize();
+
+  const int num_iiwa_positions = world_plant->num_positions();
+
+  drake::systems::DiagramBuilder<double> builder;
+  builder.AddSystem(std::move(owned_world_plant));
+  builder.AddSystem(std::move(owned_scene_graph));
+
+  auto lcm = builder.AddSystem<drake::systems::lcm::LcmInterfaceSystem>();
+
+  // Create the command subscriber and status publisher.
+  auto command_sub = builder.AddSystem(
+      drake::systems::lcm::LcmSubscriberSystem::Make<drake::lcmt_iiwa_command>(
+          "IIWA_COMMAND", lcm));
+  command_sub->set_name("command_subscriber");
+  auto command_receiver =
+      builder.AddSystem<IiwaCommandReceiver>(num_iiwa_positions);
+  command_receiver->set_name("command_receiver");
+
+  // LCM publisher system
+  const double kIiwaLcmStatusPeriod = 0.005;
+  auto iiwa_status_publisher = builder.AddSystem(
+      drake::systems::lcm::LcmPublisherSystem::Make<drake::lcmt_iiwa_status>(
+          "IIWA_STATUS", lcm, kIiwaLcmStatusPeriod /* publish period */));
+
+  // Torque Controller-- includes virtual springs and damping.
+  VectorXd stiffness, damping_ratio;
+
+  // The virtual spring stiffness in Nm/rad.
+  stiffness.resize(7);
+  stiffness << 2, 2, 2, 2, 2, 2, 2;
+
+  // A dimensionless damping ratio. See KukaTorqueController for details.
+  damping_ratio.resize(stiffness.size());
+  damping_ratio.setConstant(1.0);
+  stiffness = stiffness.replicate(1, 1);
```

Comment:
Looks like some redundant code here with stiffness and damping ratio.

Reply:
Removed redundant code.